share/vss/pedersen/vss_test.go

```diff
@@ -80,428 +81,505 @@ func TestVSSWhole(t *testing.T) {
+    assert.Equal(t, StatusComplaint, resp.Status)
+    assert.NotNil(t, v.aggregator)
+    assert.Equal(t, resp.SessionID, dealer.sid)
+
+    aggr := v.aggregator
+    r, ok := aggr.responses[uint32(v.index)]
+    assert.True(t, ok)
+    assert.Equal(t, StatusComplaint, r.Status)
+
+    // wrong index
+    resp.Index = uint32(len(verifiersPub))
+    sig, err := schnorr.Sign(suite, v.longterm, resp.Hash(suite))
+    resp.Signature = sig
+    assert.Error(t, aggr.verifyResponse(resp))
+    resp.Index = 0
+
+    // wrong signature
+    goodSig := resp.Signature
+    resp.Signature = randomBytes(len(goodSig))
+    assert.Error(t, aggr.verifyResponse(resp))
+    resp.Signature = goodSig
+
+    // wrongID
+    wrongID := randomBytes(len(resp.SessionID))
+    goodID := resp.SessionID
+    resp.SessionID = wrongID
+    assert.Error(t, aggr.verifyResponse(resp))
+    resp.SessionID = goodID
+}
+
+func TestVSSAggregatorAllResponses(t *testing.T) {
+    dealer := genDealer()
+    aggr := dealer.aggregator
+
+    for i := 0; i < aggr.t; i++ {
+        aggr.responses[uint32(i)] = &Response{Status: StatusApproval}
+    }
+    assert.True(t, aggr.EnoughApprovals())
+    assert.False(t, aggr.DealCertified())
+
+    for i := aggr.t; i < nbVerifiers; i++ {
+        aggr.responses[uint32(i)] = &Response{Status: StatusApproval}
+    }
+
+    assert.True(t, aggr.DealCertified())
+    assert.Equal(t, suite.Point().Mul(secret, nil), dealer.SecretCommit())
+}
+
+func TestVSSDealerTimeOut(t *testing.T) {
+    dealer := genDealer()
+    aggr := dealer.aggregator
+
+    for i := 0; i < aggr.t; i++ {
+        aggr.responses[uint32(i)] = &Response{Status: StatusApproval}
+    }
+
+    // Enough approvals, but all remaining responses missing
+    assert.True(t, aggr.EnoughApprovals())
+    assert.False(t, aggr.DealCertified())
+
+    // Tell dealer to consider other verifiers timed-out
+    dealer.setTimeOut()
+
+    // Deal should be certified
+    assert.True(t, aggr.DealCertified())
+    assert.NotNil(t, dealer.SecretCommit())
+}
+
+func TestVSSVerifierTimeOut(t *testing.T) {
+    dealer, verifiers := genAll()
+    v := verifiers[0]
+
+    encDeal, err := dealer.EncryptedDeal(0)
+
+    require.Nil(t, err)
+
+    // Make verifier create it's aggregator by processing EncDeal
+    resp, err := v.ProcessEncryptedDeal(encDeal)
+    require.NotNil(t, resp)
+    require.Nil(t, err)
+
+    aggr := v.aggregator
+
+    // Add t responses
+    for i := 0; i < aggr.t; i++ {
+        aggr.responses[uint32(i)] = &Response{Status: StatusApproval}
+    }
+
+    // Enough Approvals, but not a response for every verifier
+    assert.True(t, aggr.EnoughApprovals())
+    assert.False(t, aggr.DealCertified())
+
+    // Trigger time out, thus adding StatusComplaint to all
+    // remaining verifiers
+    v.setTimeOut()
+
+    // Deal must be certified now
+    assert.True(t, aggr.DealCertified())
+    assert.NotNil(t, v.Deal())
+}
+
+
+// func TestVSSAggregatorVerifyDeal(t *testing.T) {
```

Comment:
Why is this one not working ? Again, don't hesitate to add the revelant `setTimeout` call in the test to make it pass, since the test unit here want to test for the correctness of the created share, so it's not the goal to test for the timeout thing.

Reply:
This was failing because of a call to deal.RndShare instead of deal.SecShare, so probably a copy paste error from the Rabin tests. Fixed.